src/plugins/github/fetchGithubRepo.js

```diff
@@ -206,7 +206,24 @@ function tryGithubFetch(fetch, fetchOptions): Promise<any> {
 
 function retryGithubFetch(fetch, fetchOptions) {
   return new Promise((resolve, reject) => {
-    const operation = retry.operation();
+    // The retry parameters are tuned so that we will have one retry that is
+    // just after 1 hour after the initial request. This way, if we hit a
+    // GitHub rate limit exceeded error, we will be sure to have a retry after
+    // the limit has reset.
+    // In principle, we shouldn't bother having a bunch of immediate retrys
+    // when we hit rate limit exceeded, but it won't cause any problems, and the
+    // retry library does not offer any way to vary the retry period based
+    // on the specific error received.
+    //
+    // Verification that we will issue a retry after an hour:
+    // 2.8 ** 8 = 3778
+    // Minimum timeout is one second (1000 ms)
+    // 3778 seconds = 1.05 hours
```

Comment:
This shows that the delay _between_ the 7th and 8th retries is
3778 seconds. Don’t you actually want to show that the _total_ time
spent waiting between the 0th and 8th retries is just over an hour?

That is, you want to pick initial timeout `a`, ratio `r`, and retry
count `n` such that

```latex
\frac{a (r^0 - r^{n+1})}{1 - r} = 3600 + \varepsilon
```

(because the finite geometric series is just the difference of two
infinite geometric series with different initial terms).

So, with `a = 1.0` and `n = 8`, you want `(1 - r^9) / (1 - r)` to equal
(say) `3600 * 1.05`, which is satisfied for `r \approx 2.638`.

Confirm math:

```
$ ghc -e 'let { r = 2.638; a = 1.0; n = 8 } in sum [a * r^k | k <- [0..n]] / 3600'
1.049026649634996
```

Confirm API:

```
[✓] 21:38:25(1601699905) [5e0894f64d]
wchargin@wchargin-t440s ~/git/sourcecred
$ cat test.js
const retry = require("retry");
const op = retry.operation({factor: 2, retries: 3, minTimeout: 1000});
const first = Date.now();
let last = first;
op.attempt(() => {
  const now = Date.now();
  console.log(`delta: ${now - last} ms; total: ${now - first} ms`);
  last = now;
  op.retry(true);
});

[✓] 21:38:26(1601699906) [5e0894f64d]
wchargin@wchargin-t440s ~/git/sourcecred
$ node test.js
delta: 0 ms; total: 0 ms
delta: 1004 ms; total: 1004 ms
delta: 2002 ms; total: 3006 ms
delta: 4000 ms; total: 7006 ms
```

Reply:
FWIW, the error as written causes the requests to be sent at 0.58 hours past and then again at 1.63 hours past, so I do think that this matters, since you burn 38 minutes:

```
$ ghc -e 'let { r = 2.8; a = 1.0 / 3600; n = 8 } in mapM_ print $ scanl1 (+) [a * r^k | k <- [0..n]]'
2.777777777777778e-4
1.0555555555555555e-3
3.233333333333333e-3
9.33111111111111e-3
2.640488888888888e-2
7.421146666666664e-2
0.20806988444444435
0.5828734542222219
1.632323449599999
```